vms/platformvm/network/network.go

```diff
@@ -109,60 +107,81 @@ func (n *network) AppGossip(ctx context.Context, nodeID ids.NodeID, msgBytes []b
 		)
 		return nil
 	}
-
 	txID := tx.ID()
 
 	// We need to grab the context lock here to avoid racy behavior with
 	// transaction verification + mempool modifications.
+	//
+	// Invariant: tx should not be referenced again without the context lock
+	// held to avoid any data races.
 	n.ctx.Lock.Lock()
 	defer n.ctx.Lock.Unlock()
 
 	if reason := n.mempool.GetDropReason(txID); reason != nil {
 		// If the tx is being dropped - just ignore it
 		return nil
 	}
-
-	// add to mempool
-	if err := n.IssueTx(ctx, tx); err != nil {
-		n.ctx.Log.Debug("tx failed verification",
-			zap.Stringer("nodeID", nodeID),
-			zap.Error(err),
-		)
+	err = n.issueTx(tx)
+	if err == nil {
```

Comment:
I feel like this is a bit better just to make sure we never accidentally return the `err` here
```suggestion
	if err := n.issueTx(tx); err == nil {
```

Reply:
went with your suggestion [e0f1b94](https://github.com/ava-labs/avalanchego/pull/2296/commits/e0f1b944a4410caff18dc728edb35800bb259ffe)